rest/oidc_api.go

```diff
@@ -106,6 +140,18 @@ func (h *handler) handleOIDCCallback() error {
 		return base.HTTPErrorf(http.StatusUnauthorized, "oidc callback received an error: %v", errorDescription)
 	}
 
+	stateCookie, err := h.rq.Cookie(stateCookieName)
+	if stateCookie == nil {
+		return ErrNoStateCookie
+	}
+	if err != nil {
+		return ErrReadStateCookie
+	}
+	stateParam := h.rq.URL.Query().Get(requestParamState)
+	if stateCookie != nil && stateParam != stateCookie.Value {
```

Comment:
We've already nil checked stateCookie

```suggestion
	if stateParam != stateCookie.Value {
```

Reply:
Yeah, makes much sense, this additional nil check is obsolete.